### PR TITLE
Updates for Julia 0.5.

### PR DESCRIPTION
**Pull request by andreasnoack**
Import some operators instead of using `Base.(operator)` referencing for compatibility between 0.4 and 0.5

Make some array comprehensions Any such that they can be used in the `DataFrame` constructor

Adapt to shape preserving comprehensions

Some String stuff

cc: @ararslan

### CODE REVIEW

**Comment by ararslan** [comment]
+1 looks great to me. Has anyone told you yet today that you rock? 🎸 :metal: 👏

**Comment by andreasnoack** [comment]
Probably https://github.com/JuliaLang/METADATA.jl/pull/5625 would have to be merged before this one will pass.

**Comment by ararslan** [comment]
Should probably tag this one too.

File: src/abstractdataframe/io.jl [hunk 1]
```diff
@@ -8,8 +8,14 @@ function escapedprint(io::IO, x::Any, escapes::AbstractString)
     print(io, x)
 end
 
-function escapedprint(io::IO, x::AbstractString, escapes::AbstractString)
-    print_escaped(io, x, escapes)
+@static if VERSION < v"0.5.0-"
```
Comment (tkelman) [inline src/abstractdataframe/io.jl]: please be specific with version cutoffs like this for the sake of bisecting

File: test/dataframe.jl [hunk 2]
```diff
@@ -285,7 +285,7 @@ module TestDataFrame
 
     # describe
     #suppress output and test that describe() does not throw
-    devnull = @unix? "/dev/null" : "nul"
+    devnull = is_unix() ? "/dev/null" : "nul"
```
Comment (tkelman) [inline test/dataframe.jl]: minimum Compat version

**Pull request merged**